[WPE] clang-19 warning in WPEDisplay

philn opened the pull request:
#### f04cbaee78d3c48fbec1ea348a2111fd73e4ce0e
<pre>
[WPE] clang-19 warning in WPEDisplay
<a href="https://bugs.webkit.org/show_bug.cgi?id=282042">https://bugs.webkit.org/show_bug.cgi?id=282042</a>

Reviewed by Michael Catanzaro.

* Source/WebKit/WPEPlatform/wpe/WPEDisplay.cpp:
(wpeDisplayDispose): Call eglTerminate directly instead of doing it with g_clear_pointer which would
require a cast.

Canonical link: <a href="https://commits.webkit.org/285649@main">https://commits.webkit.org/285649@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/10a0f933eefef304145d9dd5170046f583efb51a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/73408 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/52837 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/26216 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/77662 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/24646 "Built successfully") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/61965 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/620 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/57676 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/16126 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/76475 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/61965 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/63154 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/38090 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/61965 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/22977 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/61965 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/20989 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/79291 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/723 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/225 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/66085 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/865 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/63166 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/65365 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16155 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/9207 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/7387 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/687 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/3424 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/717 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/701 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/719 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

philn commented on Source/WebKit/WPEPlatform/wpe/WPEDisplay.cpp:
```diff
@@ -31,6 +31,7 @@
 #include "WPEExtensions.h"
 #include "WPEInputMethodContextNone.h"
 #include <epoxy/egl.h>
+#include <epoxy/egl_generated.h>
```
Oops that one was inserted by clangd. I'll remove it before landing.

mcatanzaro reviewed (approved):
OK, but at this point it might be easier to just:

```
if (priv->eglDisplay) {
    eglTerminate(priv->eglDisplay);
    priv->eglDisplay = nullptr;
}
```

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPEDisplay.cpp:
```diff
@@ -31,6 +31,7 @@
 #include "WPEExtensions.h"
 #include "WPEInputMethodContextNone.h"
 #include <epoxy/egl.h>
+#include <epoxy/egl_generated.h>
```
Why do you need this? I think this is already included by epoxy/egl.h

philn replied:
https://github.com/WebKit/WebKit/pull/35676#discussion_r1814970132

philn committed:
[WPE] clang-19 warning in WPEDisplay
https://bugs.webkit.org/show_bug.cgi?id=282042

Reviewed by Michael Catanzaro.

* Source/WebKit/WPEPlatform/wpe/WPEDisplay.cpp:
(wpeDisplayDispose): Call eglTerminate directly instead of doing it with g_clear_pointer which would
require a cast.

Canonical link: https://commits.webkit.org/285649@main
